src/main/java/mil/dds/anet/search/mssql/MssqlLocationSearcher.java

```diff
@@ -26,6 +26,14 @@ protected void addTextQuery(LocationSearchQuery query) {
     qb.addSqlArg("containsQuery", qb.getContainsQuery(text));
   }
 
+  @Override
+  protected void addWithinPolygon(LocationSearchQuery query) {
+    qb.addWhereClause(String.format(
+        "\"uuid\" IN (SELECT \"uuid\" FROM locations WHERE "
+            + "\"geometry\".STWithin(geometry::STGeomFromText('%1$s', 3857)) = 1)",
+        query.getWithinPolygon()));
```

Comment:
Since this code is almost the same in three places, extract it into an (MS SQL specific) common method with two parameters.